### PR TITLE
Feat/1644 multi session/get session from storage all

### PR DESCRIPTION
**Pull request by NSeydoux**
This implements the different functions required to list all the known session IDs for a given storage, in order to be able to iterate through them is required. Note that `getSessionFromStorage` returns a Session object, potentially logged in, while `getSessionIdFromStorageAll` only returns the session ids, and it is up to the caller to the manually retrieve individual sessions. This is mostly for performance concerns: logging a session in requires an HTTP interaction, so if multiple sessions need to be logged in at once, that could represent a costly operation (that the server is still able to perform with the proposed API anyway). 

# Checklist

- [X] All acceptance criteria are met.
- [x] The changelog has been updated, if applicable.
- [x] New functions/types have been exported in `index.ts`, if applicable.
- [X] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: packages/browser/src/sessionInfo/SessionInfoManager.ts [hunk 1]
```diff
@@ -170,4 +170,27 @@ export class SessionInfoManager implements ISessionInfoManager {
   async clear(sessionId: string): Promise<void> {
     return clear(sessionId, this.storageUtility);
   }
+
+  /**
+   * Registers a new session, so that its ID can be retrieved.
+   * @param sessionId
+   */
+  async register(_sessionId: string): Promise<void> {
+    throw new Error("Unimplemented");
```
Comment (Vinnl) [inline packages/browser/src/sessionInfo/SessionInfoManager.ts:179]: Is this because of a shared interface for something that we don't want/need to implement in the browser?

If so: ugh... Please can we just split the two projects? :P

File: packages/node/src/ClientAuthentication.ts [hunk 2]
```diff
@@ -59,6 +59,8 @@ export default class ClientAuthentication {
     // could return to a previous redirect URL that contains OIDC params that
```
Comment (Vinnl) [inline packages/node/src/ClientAuthentication.ts:59]: Not part of this PR, but there's a comment here mentioning the browser even though this is the Node package :)

File: packages/node/src/__mocks__/ClientAuthentication.ts [hunk 3]
```diff
@@ -32,3 +45,21 @@ export const mockClientAuthentication = (): ClientAuthentication =>
...
+  logoutHandler: ILogoutHandler;
+};
+
+export const mockCustomClientAuthentication = (
+  mocks: Partial<CustomMocks>
+): ClientAuthentication =>
+  new ClientAuthentication(
+    mocks.loginHandler ?? LoginHandlerMock,
+    mocks.redirectHandler ?? RedirectHandlerMock,
+    mocks.logoutHandler ?? mockLogoutHandler(mocks.storage!),
```
Comment (Vinnl) [inline packages/node/src/__mocks__/ClientAuthentication.ts:63]: A bunch of warnings here about the non-null assertion, so either explicitly initialise it or add an ESLint ignore comment. (Likewise below.)

File: CHANGELOG.md [hunk 4]
```diff
@@ -23,6 +23,10 @@ The following sections document changes that have been released already:
 
 - `getSessionFromStorage`: a function to retrieve a session from storage based on
 its session ID (for multi-session management).
+- `getStoredSessionIdAll`: a function to retrieve the session IDs for all stored
+sessions.
+- `clearSessionAll`: a function to clear all information about all sessions in
```
Comment (Vinnl) [inline CHANGELOG.md:28]:
```suggestion
- `getSessionIdFromStorageAll`: a function to retrieve the session IDs for all stored
sessions.
- `clearSessionFromStorageAll`: a function to clear all information about all sessions in
```

**Review by Vinnl — APPROVED** [review]

File: packages/browser/src/sessionInfo/SessionInfoManager.ts (hunk 1 above)
Comment (NSeydoux) [reply packages/browser/src/sessionInfo/SessionInfoManager.ts:179]: I'm afraid so ^^.

File: packages/node/src/__mocks__/ClientAuthentication.ts (hunk 3 above)
Comment (NSeydoux) [reply packages/node/src/__mocks__/ClientAuthentication.ts:63]: Weird, ESlint did not complain (that I realised), only the GH static code analysis tool takes issue with this. I'll see how to fix this.

Comment (Vinnl) [reply packages/node/src/__mocks__/ClientAuthentication.ts:63]: I think ESLint is still only set to warnings, so it's probably somewhere in your console's logs :)

Comment (NSeydoux) [reply packages/node/src/__mocks__/ClientAuthentication.ts]: 4f6ba7c2594e2dccbf474288cb6e68734c2a6d7a

Comment (NSeydoux) [reply packages/node/src/__mocks__/ClientAuthentication.ts]: Oh yeah looking for them (in the large output of all the tests ran by lerna ^^). Should it be an error ?

Comment (Vinnl) [reply packages/node/src/__mocks__/ClientAuthentication.ts]: I think it might be good to turn warnings into an error, but I think there are so many of them that that's probably best done separately.
